syntax = "proto3";

package dns;

service DNSInfo {
    rpc InitConnection (InitRequest) returns (InitResponse) {}
    rpc InitInfo (InfoRequest) returns (InfoResponse) {}
    rpc RoundHost (HostRequest) returns (HostResponse) {}
}

message InitRequest {
    bool result = 1;
}

message InitResponse {
    bool result = 1;
}

message InfoRequest {
    string host = 1;
    int32 port = 2;
}

message InfoResponse {
    bool result = 1;
}

message HostRequest {
    bool result = 1;
}

message HostResponse {
    string host = 1;
    int32 port = 2;
}

/*
syntax = "proto3";

package helloworld;

service Counter {
    rpc SayHello (HelloRequest) returns (HelloReply) {}
}

message HelloRequest {
    string message = 1;
}

message HelloReply {
    string message = 1;
}

*/